designs/usb-light.stanza

```diff
@@ -31,15 +43,75 @@ pcb-module main-board:
   val vbus-voltage = (nom-value(recommended-voltage(property(conn.VCC.power-pin)))) as Double
   val exact-resistance = led-circuit-resistance(vbus-voltage, property(led[0].forward-voltage), property(led[0].test-current), 2)
   inst res : gen-res-cmp(closest-std-val(exact-resistance,5.0), "0603")[4]
+  val trace-thickness = 0.254
   for i in 0 to 4 do :
+    ; place parts
+    val led-i4-point = Point(-2.5, 5.0 * to-double(i) - 7.5)
+    place(led[i + 4]) at loc(led-i4-point, 90.0) on Top
+    val led-i-point = Point(-7.5, 5.0 * to-double(i) - 7.5)
+    place(led[i]) at loc(led-i-point, 90.0) on Top
+    val res-i-point = Point(2.5, 5.0 * to-double(i) - 7.5)
+    place(res[i]) at loc(res-i-point, 90.0) on Top
+    ; assign nets
     net (p5v led[i].a)
-    net (led[i + 4].a led[i].c)
-    net (res[i].p[1] led[i + 4].c)
+
+    val net-name1 = string-join(["net-led", i + 1])
+    val net-id1 = make-net( to-symbol(net-name1), [(led[i + 4].a)])
+    net (net-id1 led[i].c)
+
+    val net-name2 = string-join(["net-res", i + 1])
+    val net-id2 = make-net( to-symbol(net-name2), [(res[i].p[1])])
+    net (net-id2 led[i + 4].c)
+
     net (gnd res[i].p[2])
-    place(led[i + 4]) at loc(-2.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-    place(led[i]) at loc(-7.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-    place(res[i]) at loc(2.5, 5.0 * to-double(i) - 7.5, 90.0) on Top
-  place(conn) at loc(10.75, 0.0, 90.0) on Top
+    ; place p5v track stub
+    val led-5v-x1 = x(led-i-point) - y(center(get-pad-pose(((led[i])), "a")))
+    println("%_ %_ %_ " % [led-5v-x1 x(led-i-point)  y(center(get-pad-pose(((led[i])), "a"))) ])
+    val led-5v-y1 = y(led-i-point) 
+    geom(p5v) :
+      copper(LayerIndex(0)) = Line(trace-thickness, [ Point(led-5v-x1, led-5v-y1), Point(led-5v-x1 - 2.0, led-5v-y1) ])
+    ; place p5v track 
+      if i != 0 :
+        copper(LayerIndex(0)) = Line(trace-thickness, [Point(led-5v-x1 - 2.0, led-5v-y1) Point(led-5v-x1 - 2.0, led-5v-y1 - 5.0) ])      
+      else :
+        copper(LayerIndex(0)) = Polyline(trace-thickness, [Arc(Point(led-5v-x1 - 2.0, led-5v-y1), Point(-9.5, -9.5), Point(-8.0, -10.0))])
+
+    ; place track between the LEDs
+    val led-5v-x2 = x(led-i-point) - y(center(get-pad-pose(((led[i])), "c")))
+    val led-middle-x1 = x(led-i4-point) - y(center(get-pad-pose(((led[i + 4])), "a")))
+    println("led-5v-x2 %_" % [led-5v-x2])
+    geom(net-id1) :
+      copper(LayerIndex(0)) = Line(trace-thickness, [ Point(led-5v-x2, led-5v-y1), Point(led-middle-x1, led-5v-y1) ])
```

Comment:
Probably could add this definition earlier in the file: 
```
defn pad-center(part:JITXObject, pad-name:String) -> Point:
    center(get-pad-pose(part, pad-name))
```
Then this ought to become something like:
```suggestion
      copper(LayerIndex(0)) = Line(trace-thickness, [ pad-center(led[i], "c"), pad-center(led[i+4], "a")] )
```
But the coordinate mixing issue makes me wonder if something is going wrong here.

Reply:
I agree a higher order function would be desirable. But it belongs in utils. This proposed function doesn't address the coordinate mixing issue. I'd like to put off developing higher order functions for another PR, maybe after we've identified a few more cases.